azuread/table_azuread_application.go

```diff
@@ -35,145 +43,214 @@ func tableAzureAdApplication() *plugin.Table {
+			{Name: "identifier_uris", Type: proto.ColumnType_JSON, Description: "The URIs that identify the application within its Azure AD tenant, or within a verified custom domain if the application is multi-tenant.", Transform: transform.FromMethod("GetIdentifierUris")},
+			{Name: "info", Type: proto.ColumnType_JSON, Description: "Basic profile information of the application such as app's marketing, support, terms of service and privacy statement URLs. The terms of service and privacy statement are surfaced to users through the user consent experience.", Transform: transform.FromMethod("ApplicationInfo")},
+			{Name: "key_credentials", Type: proto.ColumnType_JSON, Description: "The collection of key credentials associated with the application.", Transform: transform.FromMethod("ApplicationKeyCredentials")},
+			{Name: "owner_ids", Type: proto.ColumnType_JSON, Hydrate: getAdApplicationOwners, Transform: transform.FromValue(), Description: "Id of the owners of the application. The owners are a set of non-admin users who are allowed to modify this object."},
+			{Name: "parental_control_settings", Type: proto.ColumnType_JSON, Description: "Specifies parental control settings for an application.", Transform: transform.FromMethod("ApplicationParentalControlSettings")},
+			{Name: "password_credentials", Type: proto.ColumnType_JSON, Description: "The collection of password credentials associated with the application.", Transform: transform.FromMethod("ApplicationPasswordCredentials")},
+			{Name: "spa", Type: proto.ColumnType_JSON, Description: "Specifies settings for a single-page application, including sign out URLs and redirect URIs for authorization codes and access tokens.", Transform: transform.FromMethod("ApplicationSpa")},
+			{Name: "tags_src", Type: proto.ColumnType_JSON, Description: "Custom strings that can be used to categorize and identify the application.", Transform: transform.FromMethod("GetTags")},
+			{Name: "web", Type: proto.ColumnType_JSON, Description: "Specifies settings for a web application.", Transform: transform.FromMethod("ApplicationWeb")},
+
+			// Standard columns
+			{Name: "tags", Type: proto.ColumnType_JSON, Description: ColumnDescriptionTags, Transform: transform.From(adApplicationTags)},
+			{Name: "title", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTitle, Transform: transform.From(adApplicationTitle)},
+			{Name: "tenant_id", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTenant, Hydrate: plugin.HydrateFunc(getTenant).WithCache(), Transform: transform.FromValue()},
 		},
 	}
 }
 
 //// LIST FUNCTION
 
 func listAdApplications(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
-	session, err := GetNewSession(ctx, d)
+	// Create client
+	client, adapter, err := GetGraphClient(ctx, d)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating client: %v", err)
```

Comment:
We should add a error logging message here, e.g.,
```
plugin.Logger(ctx).Error("azuread_application.listAdApplications", "connection_error", err)
```

Do we also need the `fmt.Errorf` message here then?

Reply:
Whatever we decide here, we should also make changes across the other functions/files.

Reply:
Right @cbruno10 that makes sense. I updated all the tables to use the same logging standard.